src/instructlab/model/backends/llama_cpp.py

```diff
@@ -203,3 +186,70 @@ def read_root():
     if queue:
         queue.close()
         queue.join_thread()
+
+
+def augment_chat_template(
+    chat_template: str,
+    model_family: str,
+    model_path: pathlib.Path,
+    queue: Optional[multiprocessing.Queue],
+):
+    if chat_template == CHAT_TEMPLATE_TOKENIZER:
+        # llama_cpp_python calculates from the tokenizer config on load, so
+        # nothing to do
+        return
+
+    if chat_template == CHAT_TEMPLATE_AUTO:
+        # Currently "auto" maps to replacing with ilab stored templates
+        template, eos_token, bos_token = get_model_template(model_family, model_path)
+    else:
+        template = load_template(chat_template)
+        eos_token = None
+        bos_token = None
+
+    logger.info(f"Replacing chat template:\n {template}")
+
+    try:
+        for proxy in llama_app.get_llama_proxy():
+            proxy().chat_handler = llama_chat_format.Jinja2ChatFormatter(
+                template=template,
+                # Use the model defined eos and bos if either is not
+                # defined (when a custom template is used)
+                eos_token=resolve_token_eos(eos_token, proxy),
+                bos_token=resolve_token_bos(bos_token, proxy),
+            ).to_chat_handler()
+    # pylint: disable=broad-exception-caught
+    except Exception as exc:
+        if queue:
+            queue.put(exc)
+            queue.close()
+            queue.join_thread()
+            return
+        raise ServerException(f"failed creating the server application: {exc}") from exc
+
+
+def resolve_token_eos(eos_token: Optional[str], proxy: LlamaProxy) -> str:
+    if eos_token is not None:
+        return eos_token
+
+    return resolve_token(proxy, proxy().token_eos())
+
+
+def resolve_token_bos(bos_token: Optional[str], proxy: LlamaProxy) -> str:
+    if bos_token is not None:
+        return bos_token
+
+    return resolve_token(proxy, proxy().token_bos())
+
+
+def resolve_token(proxy: LlamaProxy, token: int) -> str:
+    result = llama_token_get_text(proxy().model, token).decode("utf-8")
+    return cast(str, result)
+
+
+def load_template(template_path: str) -> str:
+    path = pathlib.Path(template_path)
+    if not path.exists():
+        raise FileNotFoundError("Chat template file does not exist: {}".format(path))
+
+    return path.read_text(encoding="utf-8")
```

Comment:
Have do we handle `path.read_text` failing?

Reply:
So IMO this will throw appropriate errors with the path (FileNotFound, OSError etc). Open to ideas if you have a better suggestion.

However this is a problem were we dont handle these outer sections correctly will fix that.